libchecktestdata.cc

```diff
@@ -1003,12 +1003,14 @@ void checktoken(const command& cmd)
 		if ( !isdigit(data.peek()) ) error("digit expected");
 		size_t digitpos = data.pos(), dotpos = string::npos;
 		char first_digit = data.peek();
+		size_t num_leading_digits = 0;
 		while ( (isdigit(data.peek()) ||
 		         (dotpos==string::npos && digitpos!=data.pos() && data.peek()=='.')) ) {
 			if ( data.readchar()=='.' ) dotpos = data.pos()-1;
+			else if(dotpos == string::npos) ++num_leading_digits;
```

Comment:
I would remove the `else`: it doesn't change the logic, since in the `if` clause `dotpos` would anyways be changed away from `string::npos` and I think it simplifies understanding the code.